lib/_http_agent.js

```diff
@@ -46,33 +46,31 @@ function Agent(options) {
 
   EventEmitter.call(this);
 
-  var self = this;
+  this.defaultPort = 80;
+  this.protocol = 'http:';
 
-  self.defaultPort = 80;
-  self.protocol = 'http:';
-
-  self.options = util._extend({}, options);
+  this.options = util._extend({}, options);
 
   // don't confuse net and make it think that we're connecting to a pipe
-  self.options.path = null;
-  self.requests = {};
-  self.sockets = {};
-  self.freeSockets = {};
-  self.keepAliveMsecs = self.options.keepAliveMsecs || 1000;
-  self.keepAlive = self.options.keepAlive || false;
-  self.maxSockets = self.options.maxSockets || Agent.defaultMaxSockets;
-  self.maxFreeSockets = self.options.maxFreeSockets || 256;
-
-  self.on('free', function(socket, options) {
-    var name = self.getName(options);
+  this.options.path = null;
+  this.requests = {};
+  this.sockets = {};
+  this.freeSockets = {};
+  this.keepAliveMsecs = this.options.keepAliveMsecs || 1000;
+  this.keepAlive = this.options.keepAlive || false;
+  this.maxSockets = this.options.maxSockets || Agent.defaultMaxSockets;
+  this.maxFreeSockets = this.options.maxFreeSockets || 256;
+
+  this.on('free', (socket, options) => {
```

Comment:
IIRC this function could be extracted out of the constructor entirely and just merely referenced here, since I believe it only uses `self` from the parent/constructor scope.